libnmstate/nm/wired.py

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate import nmclient
+
+
+def get_info(active_connection):
+    """
+    Provides the current active values for an active connection.
+    """
+    info = {}
+    # FIXME: Maybe use
```

Comment:
Looks like we fetch the "desired" config, is it assured to be the actual? I was wandering the same for the other info we collect.

Reply:
This might be something that we need to address in general. Currently we are using `dev.get_active_connection()` but from a discussion with @thom311 I understood that the active connection is only a pointer and might show to a configuration that is not yet, applied. But Using `dev.get_applied_connection(0)` would return the live configuration.

In general I think I would wish that there was a clear settings interface as a class that settings such as for the wired parameters could inherit. Then the parent class could provide an interface to the connection that would need to be changed or inspected to return the actual info. Maybe just provide the connection to the constructor and have two methods apply and show, where apply gets the nmstate to apply and show returns the partial state with the settings for the respective NM setting.